### ISSUE TITLE
我无法使用ChatGLMForSequenceClassification进行分类

### ISSUE THREAD

**Issue by Mr-Lnan** [THUDM/GLM-4#220]
### System Info / 系統信息

Cuda 12.0
torch 2.0.1
transformers 4.41.2

### Who can help? / 谁可以帮助到您？

@zr

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [X] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

`model_path = "/liukeshuang/test/keshuang/ChatGLM4/glm4-9b"
model = AutoModelForSequenceClassification.from_pretrained(model_path,num_labels=3, trust_remote_code=True,device_map="auto",torch_dtype=torch.bfloat16)
tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True,torch_dtype=torch.bfloat16)
# 加载 label lora权重
model = PeftModel.from_pretrained(model, './output/label_xfg/checkpoint-10000')
model = model.eval()`
<img width="665" alt="image" src="https://github.com/THUDM/GLM-4/assets/101647003/b0bebbc4-8544-458e-8a59-5f2018ba4c55">
<img width="615" alt="image" src="https://github.com/THUDM/GLM-4/assets/101647003/321420f0-064e-4731-a603-662f61ea1cf9">
<img width="657" alt="image" src="https://github.com/THUDM/GLM-4/assets/101647003/4c016a58-996b-4192-a260-0c41c4aa030a">

### Expected behavior / 期待表现

我该怎么解决，是我的代码有问题吗

**Comment by zRzRzRzRzRzRzR** [comment]
者应该是一个bug，我们会处理

**Comment by duzx16** [comment]
修了 https://huggingface.co/THUDM/glm-4-9b/commit/4ec3b653a3c9916e0009224557e12ce9edfa02d1
另外 `input_ids` 和 `attention_masks` 不要用 bfloat16，用默认的 dtype 就行

**Comment by ljsun** [comment]
@Mr-Lnan 请教下，你在使用ChatGLMForSequenceClassification时，会出现output.logits 为nan的情况吗

**Comment by YangQianli92** [comment]
> > @Mr-Lnan请教下，你在使用ChatGLMForSequenceClassification时，会出现output.logits为nan的情况吗
> 
> 我也是遇到了这个问题，目前还没有解决

求问，解决了吗？

**Comment by fjjF77** [comment]
我也遇到了这个问题。
调试发现似乎是因为应用ChatGLMForSequenceClassification类时，会自动添加一个分类头，但是其参数没有初始化，导致可能存在nan的情况。
但是我修改代码初始化参数之后，发现训练1轮过后，梯度回传后，参数再次变成nan了。请问如何解决？
